### PR TITLE
Refactor: Implement Monorepo

### LINKED ISSUE #71: monorepo: multiple test-plans implementation in a monorepo
We need to think about either it is viable to start creating `test-infra` into a monorepo that contains all test-plans implementation

### PR DESCRIPTION
**Pull request by derrandz**
# Overview

We desire to have a monorepo setup to allow us to host multiple testplans in isolation, each as their own go module.
This PR propose a first iteration towards a fully fledge testplans monorepo/

# Reference Issue(s)
Closes #71 

# Notable Changes

- [x] Moved existing testplans under `./plans/NAME`
- [x] Factored `testkit` as its own go module
- [x] Introduced a `Makefile` to provide concise commands respecting the repositories conventions.
- [x] Updated the `README.md` to include relevant documentation.

# Additional Info

Since we have separated `testkit` into its own go module, and since test plans are "copied" by testground to `$TESTGROUND_HOME` thus disabling the ability to use relativce paths to retrieve `testkit`, testplans -as a result- have to "require" specific versions of `testkit`, and we are currently doing so using commit hashes.

So every new important change to the `testkit` module has to be accompanied with updates to `go.mod` files in every test plans that is interested in using those new changes.

This process is documented in the `Testkit` caveats in the README.md file of the repository.

### CODE REVIEW

**Comment by Bidon15** [comment]
You can mention a word 'Closes #<issue-id>'  in description to link and close automatically once this PR is merged 👍

**Review by Bidon15 — CHANGES_REQUESTED** [review]
Awesome 🎈 🎉

File: testkit/sync.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package testkit
+package sdk
```
[the remark this answers is not in the document]
Comment (derrandz) [reply testkit/sync.go:1]: Thanks for catching this!
This is a remanent (_that I did not catch_) of when the folder was named `sdk`, so it's the second option.

File: plans/001-big-blocks/tests/app-sync/run_seeds.go [hunk 2]
```diff
@@ -63,7 +63,7 @@ func RunSeed(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 	}
 
 	initGenCh := make(chan string)
-	sub, err := syncclient.Subscribe(ctx, testkit.InitialGenenesisTopic, initGenCh)
+	sub, err := syncclient.Subscribe(ctx, sdk.InitialGenenesisTopic, initGenCh)
```
Comment (Bidon15) [inline plans/001-big-blocks/tests/app-sync/run_seeds.go:66]: oops 😅 
I suppose we need to change back to testkit to make it work, isn't it?

**Review by Bidon15 — CHANGES_REQUESTED** [review]
there is some housekeeping work with changing back to testkit from sdk and we can merge 👍

File: plans/001-big-blocks/tests/app-sync/run_seeds.go (hunk 2 above)
Comment (derrandz) [reply plans/001-big-blocks/tests/app-sync/run_seeds.go:66]: 😅

File: testkit/nodekit/node.go [hunk 3]
```diff
@@ -38,7 +38,7 @@ func NewNode(
 ) (*nodebuilder.Node, error) {
 	// This is necessary to ensure that the account addresses are correctly prefixed
 	// as in the celestia application.
-	sdkcfg := sdk.GetConfig()
+	sdkcfg := testkit.GetConfig()
```
Comment (Bidon15) [inline testkit/nodekit/node.go:41]:
```suggestion
	sdkcfg := sdk.GetConfig()
```
Oops 😅  - we need to keep sdk here as it is a mention to `sdk "github.com/cosmos/cosmos-sdk/types"` import above 


image

**Review by Bidon15 — COMMENTED** [review]
Let's hit `One more` commit and we morge 

image

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+## tg-create-testplan: Create test plan under ./plans/ of this repository
+tg-create-testplan: check-name-arg
+	TESTGROUND_HOME=${DIR_FULLPATH} testground plan create --plan=${NAME}
+	@rm -rf ./data ./sdks
+	@mkdir ./docs/test-plans/${NAME}
+	@cp ./docs/test-plans/tp-template.md ./docs/test-plans/${NAME}/${NAME}.md
+.PHONY: tg-create-testplan
+
+## tg-import-testplan Import testplan to TESTGROUND_HOME
+tg-import-testplan: check-testplan-arg check-name-arg
```
[the remark this answers is not in the document]
Comment (Bidon15) [reply Makefile:114]: Ah, nvm. 
Got the idea now. We need to add more documentation to this

**Review by Bidon15 — APPROVED** [review]
LGTM
We need to think about improving makefile documentation in a separate issue/PR 

Thanks for the work @derrandz 👏 
:shipit:

File: Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (derrandz) [reply Makefile:114]: Let's discuss this some time , I get the confusing and it's justified.

**Pull request merged**